modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,239 @@
+            patient_phone_number: request[:phone_number],
+            patient_email: request[:email],
+            best_time_to_call: request[:best_timeto_call],
+            friendly_location_name: request[:friendly_location_name]
+          )
+        end
+        # rubcop:enable Metrics/MethodLength
+
+        def appointment_type(request, klass)
+          # this is temporary because test data does not include video type
+          unless request[:visit_type].in?(['Office Visit', 'Express Care', 'Phone Call'])
+            log_message_to_sentry('Unknown appointment request type', :error, { visit_type: request[:visit_type] })
+          end
+          klass::APPOINTMENT_TYPE
+        end
+
+        def phone_only?(request)
+          request[:visit_type] == 'Phone Call'
+        end
+
+        def proposed_times(request)
+          {
+            option_date1: request[:option_date1] == 'No Date Selected' ? nil : request[:option_date1],
+            option_time1: request[:option_time1] == 'No Time Selected' ? nil : request[:option_time1],
+            option_date2: request[:option_date2] == 'No Date Selected' ? nil : request[:option_date2],
+            option_time2: request[:option_time2] == 'No Time Selected' ? nil : request[:option_time2],
+            option_date3: request[:option_date3] == 'No Date Selected' ? nil : request[:option_date3],
+            option_time3: request[:option_time3] == 'No Time Selected' ? nil : request[:option_time3]
+          }
+        end
+
+        # this is used for sorting requests into appointments list
+        # if all proposed times times are in the past, use the chronologically first proposed time
+        # if any proposed times are in the future, use the one that occurs soonest
+        # because we only have date an AM/PM, setting to 8AM and 12PM because when creating appointment
+        def localized_start_date(request)
+          time_zone = time_zone(request) || '+00:00'
+          proposed_times = (1..3).each_with_object([]) do |i, results|
+            date = request["option_date#{i}"]
+            next if date.in?([nil, 'No Date Selected'])
+
+            month, day, year = date.split('/').map(&:to_i)
+            hour = request["option_time#{i}"] == 'AM' ? 8 : 12
+
+            results << DateTime.new(year, month, day, hour, 0).in_time_zone(time_zone)
+          end.sort
+
+          current_time = Time.current.in_time_zone(time_zone)
+          future_times = proposed_times.select { |time| time >= current_time }
+          future_times.any? ? future_times.first : proposed_times.first
+        end
+
+        def status(request)
+          request[:status].upcase
+        end
+
+        def status_detail(request)
+          return nil unless request[:status] == 'Cancelled'
+
+          first_detail = request.dig(:appointment_request_detail_code, 0)
+          cancellation_code = first_detail&.dig(:detail_code, :code)
+          return nil unless cancellation_code
```

Comment:
Depending on how the front end uses this data, it might be better to set this to 'Cancelled - other'.